### PR TITLE
#1438 BundlesTest can now fetch resources from subdirectories

### PR DESCRIPTION
**Pull request by carlosmiranda**
#1438: Fixed bug where `BundlesTest` could not fetch resources from the bundle subdirectories.

The problem was that it tried to get the XML path relative to the bundle path. However, by checking `lastIndexOf('/')`, it ended up eating the subdirectory path as well:

* `com/zerocracy/bundles/fetches_subdirectories/options/foo.xml` -> `foo.xml`

This was fixed so that the relative path is now checked against not just `/` but the entire bundle path (e.g. `com/zerocracy/bundles/fetches_subdirectories/`

* `com/zerocracy/bundles/fetches_subdirectories/options/foo.xml` -> `options/foo.xml`
 
Also added a bundles test to see if it can get project file contents both from the bundle home directory and subdirectory.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1471](https://github.com/zerocracy/farm/issues/1471) is now in scope, role is `REV`

File: src/test/resources/com/zerocracy/bundles/fetches_subdirectories/_after.groovy [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.zerocracy.bundles.graduation
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/fetches_subdirectories/_after.groovy]: @carlosmiranda wrong package name

File: src/test/resources/com/zerocracy/bundles/fetches_subdirectories/_before.groovy [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.zerocracy.bundles.graduation
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/fetches_subdirectories/_before.groovy]: @carlosmiranda package name should be `...fetches_subdirectories`

**Comment by carlosmiranda** [comment]
@krzyk fixed, please check again

**Comment by krzyk** [comment]
@carlosmiranda thanks

**Comment by krzyk** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1471](https://github.com/zerocracy/farm/issues/1471) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@krzyk According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by krzyk** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@krzyk thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by ypshenychka** [comment]
@g4s8 I wasn't paid as well.

**Comment by g4s8** [comment]
@0crat pay 15min @krzyk

**Comment by g4s8** [comment]
@0crat pay 8min @ypshenychka

**Comment by g4s8** [comment]
@krzyk @ypshenychka it's duplicate claim issue again, I'm working on migrating to SQS queues right now, and then we should fix that duplicate claims issue.

**Comment by 0crat** [comment]
Direct payment from @g4s8[/z](https://www.0crat.com/u/g4s8): +15 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Direct payment to @krzyk[/z](https://www.0crat.com/u/krzyk) in #1471, which is discouraged, see [§49](http://www.zerocracy.com/policy.html#49): -20 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
Direct payment from @g4s8[/z](https://www.0crat.com/u/g4s8): +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Direct payment to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) in #1471, which is discouraged, see [§49](http://www.zerocracy.com/policy.html#49): -20 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@g4s8 Does it mean I shouldn't give any QA verdicts until this migration is finished?

**Comment by g4s8** [comment]
@ypshenychka no, but this error may occur sometimes

**Pull request merged**